components/FormCollection/FormCollection.js

```diff
@@ -0,0 +1,138 @@
+import styled, { keyframes, css } from "styled-components";
+import RegularButton from "../Buttons/RegularButton";
+import ButtonWrapper from "../Buttons/ButtonWrapper";
+import FormInput from "../FormFlashcard/FormInput";
+
+export default function FormCollection({
+  headline,
+  actionMode,
+  changeActionMode,
+  currentCollection,
+  isFormClosing,
+  onEditCollection,
+}) {
+  async function handleSubmit(event) {
+    event.preventDefault();
+
+    const formData = Object.fromEntries(new FormData(event.target));
+    const { collectionName, collectionColor } = formData;
+
+    let updatedCollection = {
+      ...currentCollection,
+      title: collectionName,
+      color: collectionColor,
+    };
+
+    try {
+      await onEditCollection(updatedCollection);
+      event.target.reset();
+      changeActionMode("default");
+    } catch (error) {
+      console.error("Error while editing collection:", error);
```

Comment:
Additionally you could add user feedback (e.g., setError or error state for UI rendering).

Reply:
Hey, I already have error handling for edit Collection in the _App.js.